spec/models/category_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe Category, type: :model do
+  describe 'associations' do
+    it { should belong_to(:user) }
+    it { should have_many(:budget_transactions_categories) }
+    it { should have_many(:budget_transactions).through(:budget_transactions_categories) }
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of(:name) }
+    it { should validate_presence_of(:icon) }
+  end
+
+  describe 'total_amount' do
```

Comment:
- [ ] While you have done good job writing some tests, I will encourage you to write sufficient test cases for your project as it requires unit and integration tests for all the most important components of your RoR application but I have seen scanty test. This is to help your app ensure it works as you want it and also help you learn equally.
![image](https://github.com/ishimwepati/Budget-app/assets/86200356/10dcc508-a275-4d26-83fd-eaf89a88e1ed)

Reply:
Done

Reply:
* As mentioned by the previous CR this is a capstone project and it requires that you write tests for all important components, Kindly add more tests both unit and integration tests 🙏

Reply:
🤔🤔🤔🤔🤔

Reply:
Kindly do this and you'll see all the Test !

<img width="698" alt="Screenshot 2023-12-10 at 01 34 41" src="https://github.com/ishimwepati/Budget-app/assets/14818338/89ee7375-66de-4c42-8b93-6ee9eec8ddda">

Reply:
- Your tests are very few, you should kindly add more so that they are not scanty.

Reply:
- [ ] Again here as requested by both previous reviewers kindly add some tests as this is a requirement. You could choose to add unit tests.

Here is an example of unit tests for the categories index page

```
# spec/views/categories/index.html.erb_spec.rb

require 'rails_helper'

RSpec.describe 'categories/index.html.erb', type: :view do
  before do
    assign(:category_info, { icon: 'icon_path', name: 'Category Name' })
    assign(:total_amount, 100.0)
    assign(:budget_transactions, [double(name: 'Transaction Name', amount: 50.0)])
  end

  it 'displays the category information' do
    render
    expect(rendered).to have_content('TRANSACTIONS')
    expect(rendered).to have_content('Category Name')
    expect(rendered).to have_content('$100.00')
  end

  it 'displays budget transactions' do
    render
    expect(rendered).to have_content('Transaction Name')
    expect(rendered).to have_content('$50.00')
  end

  it 'displays the "Add Transaction" link' do
    render
    expect(rendered).to have_link('Add Transaction', href: new_category_budget_transaction_path(instance_of(Category)))
  end
end
```